source/base/data_out_base.cc

```diff
@@ -7544,6 +7544,81 @@ DataOutInterface<dim, spacedim>::write_pvtu_record(
 }
 
 
+template <int dim, int spacedim>
+std::string
+DataOutInterface<dim, spacedim>::write_vtu_with_pvtu_record(
+  const std::string &folder,
+  const std::string &filename_without_extension,
+  const unsigned int counter,
+  const unsigned int n_digits_counter,
+  const MPI_Comm &   mpi_communicator,
+  const unsigned int n_groups) const
+{
+  const unsigned int rank = Utilities::MPI::this_mpi_process(mpi_communicator);
+  const unsigned int n_ranks =
+    Utilities::MPI::n_mpi_processes(mpi_communicator);
+  const unsigned int n_files_written =
+    (n_groups == 0 || n_groups > n_ranks) ? n_ranks : n_groups;
+
+  const unsigned int n_digits =
+    static_cast<int>(std::ceil(std::log10(std::fabs(n_files_written))));
+
+  const unsigned int color = rank % n_files_written;
+  const std::string  filename =
+    folder + filename_without_extension + "_" +
+    Utilities::int_to_string(counter, n_digits_counter) + "." +
+    Utilities::int_to_string(color, n_digits) + ".vtu";
+
+  if (n_groups == 0 || n_groups > n_ranks)
+    {
+      // every processor writes one file
+      std::ofstream output(filename.c_str());
+      this->write_vtu(output);
+    }
+  else if (n_groups == 1)
+    {
+      // write only a single data file in parallel
+      this->write_vtu_in_parallel(filename.c_str(), mpi_communicator);
+    }
+  else
+    {
+#ifdef DEAL_II_WITH_MPI
+      // write n_groups data files
+      MPI_Comm comm_group;
+      MPI_Comm_split(mpi_communicator, color, rank, &comm_group);
```

Comment:
Here and a couple of lines below, we ought to check the return code. We have `AssertThrowMpi` for this.

Reply:
done in https://github.com/dealii/dealii/pull/8904/commits/554bef32cf65c39b12892a61e937ffebf78e3c1a